pocs/tests/utils/test_utils.py

```diff
@@ -51,3 +52,30 @@ def test_has_camera_ports():
 
     for port in ports:
         assert port.startswith('usb:')
+
+
+def test_countdown_timer_bad_input():
+    with pytest.raises(ValueError):
+        assert CountdownTimer('d')
+
+    with pytest.raises(ValueError):
+        assert CountdownTimer(current_time())
+
+    with pytest.raises(AssertionError):
+        assert CountdownTimer(-1)
+
+
+def test_countdown_timer_non_blocking():
+    timer = CountdownTimer(0)
+    assert timer.is_non_blocking
+    assert timer.time_left() == 0
+
+
+def test_countdown_timer():
+    timer = CountdownTimer(1)
+    assert timer.time_left() > 0
+    assert timer.expired() is False
+    assert timer.is_non_blocking is False
+    time.sleep(1)
```

Comment:
I'd prefer a sleep time that is greater than the count down time. Alternately, you could have a loop that checks every 0.01 seconds (for example), and tracks whether the time_left is decreasing each time, and goes to zero before 1.1 seconds (for example) has passed.

Reply:
Good idea. I update this, let me know if you think still needs improvement.